### PR TITLE
Add IE11 polyfill for `new Set(iterable)`

### PR DESCRIPTION
**Pull request by alanorozco**
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set/Set#browser_compatibility

### CODE REVIEW

**Review by kristoferbaxter — APPROVED** [review]
Let's get another reviewer too, but this LGTM.

File: build-system/test-configs/conformance-config.textproto [hunk 1]
```diff
@@ -261,12 +261,6 @@ requirement: {
 
 # Set
 
-requirement: {
-  type: BANNED_CODE_PATTERN
-  error_message: 'Cannot construct Set with Iterable'
-  value: '/** @param {Array|Object} a */function template(a) {new Set(a)}'
-}
-
```
Comment (alanorozco) [inline build-system/test-configs/conformance-config.textproto:269]: Odd that this didn't prevent existing uses.
```
extensions/amp-image-lightbox/0.1/amp-image-lightbox.js:
  50: const SUPPORTED_ELEMENTS_ = new Set(['amp-img', 'amp-anim', 'img']);
extensions/amp-image-slider/0.1/amp-image-slider.js:
  35: const VALID_IMAGE_TAGNAMES = new Set(['AMP-IMG', 'IMG']);
extensions/amp-image-viewer/0.1/amp-image-viewer.js:
  62: const ELIGIBLE_TAGS = new Set(['AMP-IMG', 'AMP-ANIM', 'IMG']);
extensions/amp-lightbox-gallery/0.1/service/lightbox-manager-impl.js:
  42: const LIGHTBOX_ELIGIBLE_TAGS = new Set(['AMP-IMG', 'IMG']);
  43  // ... (and more similar uses)
extensions/amp-pan-zoom/0.1/amp-pan-zoom.js:
  51: const ELIGIBLE_TAGS = new Set([
  52    'svg',
```

File: src/polyfills/set.js [hunk 2]
```diff
@@ -14,19 +14,35 @@
...
+  const s = new SetConstructor([1]);
+  // Add suppport for `new Set(iterable)`. IE11 lacks it.
+  if (s.size < 1) {
+    win.Set = /** @type {typeof Set} */ (
+      function (iterable) {
+        const set = new SetConstructor();
+        if (iterable) {
+          const asArray = toArray(iterable);
+          for (let i = 0; i < asArray.length; i++) {
+            set.add(asArray[i]);
```
Comment (jridgewell) [inline src/polyfills/set.js:33]: In order to support iterables, and not just array-likes.

```suggestion
          for (const it of iterable) {
            set.add(it);
```

File: src/polyfills/index.js [hunk 3]
```diff
@@ -43,7 +43,7 @@ if (!IS_ESM) {
   installArrayIncludes(self);
   installMapSet(self);
   installWeakMapSet(self);
-  installSetAdd(self);
+  installSet(self);
```
Comment (jridgewell) [inline src/polyfills/index.js:46]: Strange that this list isn't kept in sync with https://github.com/ampproject/amphtml/blob/33dd4282678b748014ceea1424bc7134c482c93e/src/friendly-iframe-embed.js#L698-L701.

**Review by jridgewell — APPROVED** [review]

**Pull request merged**